micro/DRAM.cpp

```diff
@@ -0,0 +1,75 @@
+#include "common.h"
+
+namespace s = cl::sycl;
+
+template <typename DataT>
+class MicroBenchDRAMKernel;
+
+template <typename DataT>
+size_t getBufferSize(size_t problemSizeMiB) {
+  return (problemSizeMiB * 1024 * 1024) / sizeof(DataT);
+}
+
+/**
+ * Microbenchmark measuring DRAM throughput.
+ */
+template <typename DataT>
+class MicroBenchDRAM {
+protected:
+  BenchmarkArgs args;
+  const size_t buffer_size;
+  std::vector<DataT> input;
+  s::buffer<DataT, 1> input_buf;
+  s::buffer<DataT, 1> output_buf;
+
+public:
+  MicroBenchDRAM(const BenchmarkArgs& _args)
+      : args(_args), buffer_size(getBufferSize<DataT>(args.problem_size)), input_buf(s::range<1>(buffer_size)),
+        output_buf(s::range<1>(buffer_size)) {}
+
+  void setup() {
+    input.resize(buffer_size, 33.f);
+    args.device_queue
+        .submit([&](cl::sycl::handler& cgh) {
+          auto acc = input_buf.template get_access<s::access::mode::discard_write>(cgh);
+          cgh.copy(input.data(), acc);
+        })
+        .wait();
```

Comment:
I would suggest to move this `wait()` to the benchmark framework in `common.h` before it executes `run()`. I suspect other benchmarks may forget that synchronization is necessary here, so it's better to ensure globally that before `run()` is executed, the queue is empty.

Reply:
Alright, will do!